### PR TITLE
Remove versioning of MQTT CLI

### PR DESCRIPTION
**Pull request by LukasHiveMQ**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -133,17 +134,23 @@ val swarmOpenApi: Configuration by configurations.creating {
     }
 }
 
+val swarmOpenApiUpdater: Configuration by configurations.creating {
```
Comment (hurtadosanti) [inline build.gradle.kts:137]: Seems to me that a function can be created to avoid duplication

File: build.gradle.kts [hunk 2]
```diff
@@ -133,17 +134,23 @@ val swarmOpenApi: Configuration by configurations.creating {
...
+    attributes {
+        attribute(Category.CATEGORY_ATTRIBUTE, objects.named("open-api"))
+    }
+}
+
 dependencies {
+    hivemqOpenApi(files("specs/hivemq-openapi.yaml"))
+    swarmOpenApi(files("specs/hivemq-swarm-openapi.yaml"))
+
     if (gradle.includedBuilds.any { it.name == "hivemq-enterprise" }) {
```
Comment (hurtadosanti) [inline build.gradle.kts:149]: Can you comment why are you having this if/else set

File: gradle.properties [hunk 3]
```diff
@@ -1,5 +1,9 @@
-version=4.8.0-SNAPSHOT
```
Comment (hurtadosanti) [inline gradle.properties:1]: have the dependencies being check for vulnerabilities?

File: build.gradle.kts [hunk 4]
```diff
@@ -601,8 +608,9 @@ tasks.startShadowScripts {
 /* ******************** HiveMQ composite build ******************** */
 
 tasks.register<Sync>("updateSpecs") {
-    from(hivemqOpenApi) { rename { "hivemq-openapi.yaml" } }
-    from(swarmOpenApi) { rename { "hivemq-swarm-openapi.yaml" } }
+    group = "cli"
```
Comment (SgtSilvio) [inline build.gradle.kts:611]: Every tasks here is part of the cli.
```suggestion
    group = "openapi"
```

File: build.gradle.kts [hunk 5]
```diff
@@ -601,8 +608,9 @@ tasks.startShadowScripts {
 /* ******************** HiveMQ composite build ******************** */
 
 tasks.register<Sync>("updateSpecs") {
```
Comment (SgtSilvio) [inline build.gradle.kts:610]: we should probably also rename this to updateOpenApiSpecs

File: build.gradle.kts (hunk 2 above)
Comment (SgtSilvio) [inline build.gradle.kts:149]: the if is not necessary at all here

File: build.gradle.kts [hunk 6]
```diff
@@ -133,17 +134,23 @@ val swarmOpenApi: Configuration by configurations.creating {
...
+    swarmOpenApi(files("specs/hivemq-swarm-openapi.yaml"))
+
     if (gradle.includedBuilds.any { it.name == "hivemq-enterprise" }) {
-        hivemqOpenApi("com.hivemq:hivemq-enterprise")
-    } else {
-        hivemqOpenApi(files("specs/hivemq-openapi.yaml"))
+        hivemqOpenApiUpdater("com.hivemq:hivemq-enterprise")
     }
-
     if (gradle.includedBuilds.any { it.name == "hivemq-swarm" }) {
```
Comment (SgtSilvio) [inline build.gradle.kts:152]: if not necessary

File: build.gradle.kts [hunk 7]
```diff
@@ -125,6 +118,14 @@ val hivemqOpenApi: Configuration by configurations.creating {
     }
```
Comment (SgtSilvio) [inline build.gradle.kts:118]: The attributes here are not necessary currently as we are loading them now only from files.
We could make a "repository" out of the specs folder with proper metadata, but that's something for a different ticket.

File: build.gradle.kts [hunk 8]
```diff
@@ -125,6 +118,14 @@ val hivemqOpenApi: Configuration by configurations.creating {
...
 }
 
+val hivemqOpenApiUpdater: Configuration by configurations.creating {
```
Comment (SgtSilvio) [inline build.gradle.kts:121]: The naming is weird, this is a configuration not an updater (you can not call update on it and then it performs an update).
Maybe we call it hivemqOpenApiFromProject

File: build.gradle.kts (hunk 1 above)
Comment (SgtSilvio) [inline build.gradle.kts:137]: same weird naming here

Comment (SgtSilvio) [reply build.gradle.kts:137]: Functions in build scripts actually are not the best idea regarding maintenance.
Anyhow, we do not need this 4 times as I wrote in a comment above

File: gradle.properties (hunk 3 above)
Comment (LukasHiveMQ) [reply gradle.properties:1]: I did not change any dependency version just moved the version of the project itself to the build.gradle.kts file

File: build.gradle.kts [hunk 9]
```diff
@@ -598,16 +603,20 @@ tasks.startShadowScripts {
...
+/* ******************** version updating ******************** */
 
-tasks.register<Sync>("updateSpecs") {
-    from(hivemqOpenApi) { rename { "hivemq-openapi.yaml" } }
-    from(swarmOpenApi) { rename { "hivemq-swarm-openapi.yaml" } }
-    into("specs")
-}
+val updateVersionInFiles by tasks.registering {
+    group = "version"
+    val filesToUpdate = arrayOf("docs/_docs/installation.md")
```
Comment (SgtSilvio) [inline build.gradle.kts:610]:
```suggestion
    val filesToUpdate = files("docs/_docs/installation.md")
```

File: build.gradle.kts [hunk 10]
```diff
@@ -598,16 +603,20 @@ tasks.startShadowScripts {
...
 
-plugins.withId("com.hivemq.version-updater") {
-    project.ext.set("versionUpdaterFiles", arrayOf("docs/_docs/installation.md"))
+    doLast {
+        filesToUpdate.forEach {
+            val file = project.file(it)
+            val text = file.readText()
+            val replacedText = text.replace("${prevVersion}(-SNAPSHOT)?".toRegex(), project.version.toString())
+            file.writeText(replacedText)
```
Comment (SgtSilvio) [inline build.gradle.kts:617]:
```suggestion
            val text = it.readText()
            val replacedText = text.replace("${prevVersion}(-SNAPSHOT)?".toRegex(), project.version.toString())
            it.writeText(replacedText)
```

**Review by SgtSilvio — APPROVED** [review]

**Pull request merged**